### PR TITLE
Fix bazel 5.x support for now

### PR DESCRIPTION
**Pull request by keith**
This hack allows us to fake a bazel version check for now.

### CODE REVIEW

File: apple/internal/linking_support.bzl [hunk 1]
```diff
@@ -157,12 +157,20 @@ def _register_linking_action(
         linkopts.extend(["-bundle_loader", bundle_loader_file.path])
         link_inputs.append(bundle_loader_file)
 
+    # TODO: This is a hack to support bazel 5.x and 6.x at the same time after
+    # should_lipo was removed from the arguments list, but is still required
+    # before that point. The addition of link_multi_arch_static_library probably
+    # doesn't line up perfectly, but should be good enough
```
Comment (brentleyjones) [inline apple/internal/linking_support.bzl:163]:
```suggestion
    # doesn't line up perfectly, but should be good enough.
```

File: apple/internal/linking_support.bzl [hunk 2]
```diff
@@ -157,12 +157,20 @@ def _register_linking_action(
         linkopts.extend(["-bundle_loader", bundle_loader_file.path])
         link_inputs.append(bundle_loader_file)
 
+    # TODO: This is a hack to support bazel 5.x and 6.x at the same time after
```
Comment (brentleyjones) [inline apple/internal/linking_support.bzl:160]: I assume we want to create an issue to have this removed once we remove support for 5.x?

Comment (keith) [reply apple/internal/linking_support.bzl:160]: I'll submit a revert as soon as this is in

**Review by thii — APPROVED** [review]

File: apple/internal/linking_support.bzl (hunk 2 above)
Comment (keith) [reply apple/internal/linking_support.bzl:160]: https://github.com/bazelbuild/rules_apple/pull/1386

**Pull request merged**